files/manifests/k8s-api-server.yaml

```diff
@@ -42,7 +42,11 @@ spec:
     - --enable-admission-plugins=NamespaceLifecycle,LimitRanger,ServiceAccount,ResourceQuota,DefaultStorageClass,PersistentVolumeClaimResize,PodSecurityPolicy,Priority,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook
     - --cloud-provider={{.Cluster.Kubernetes.CloudProvider}}
     - --service-cluster-ip-range={{.Cluster.Kubernetes.API.ClusterIPRange}}
-    - --etcd-servers=https://127.0.0.1:2379
+    {{- if .Etcd.HighAvailability -}}
+    - --etcd-servers=https://{{ .Cluster.Etcd.Domain }}:2379
+    {{- else -}}
+    - --etcd-servers=https://{{ .Cluster.Etcd.Domain }}:2379
+    {{- end -}}
```

Comment:
These are the same. 

(Is there any reason _not_ to do the same on non-HA?)

Reply:
> These are the same.

-_-

> (Is there any reason _not_ to do the same on non-HA?)

yes. On azure we can't do this. Azure LBs are network load balancers. Traffic leaving the node and coming back to the same node will have src IP == dst IP and the kernel drops the packet ("martian source" error). So we simply can't do the same there

Reply:
Damn that sucks!